src/main/scala/xiangshan/backend/MemBlock.scala

```diff
@@ -1675,7 +1681,78 @@ class MemBlockInlinedImp(outer: MemBlockInlined) extends LazyModuleImp(outer)
         lsq.io.exceptionAddr.vaddr
       )
     )
-  ))
+  )
+  val exceptionVaNeedExt = Mux(
+    atomicsException,
+    false.B,
+    Mux(misalignBufExceptionOverwrite,
+      true.B,
+      Mux(vSegmentException,
+        false.B,
+        lsq.io.exceptionAddr.vaNeedExt
+      )
+    )
+  )
+  val exceptionIsHyper = Mux(
+    atomicsException,
+    false.B,
+    Mux(misalignBufExceptionOverwrite,
+      misalignBufExceptionIsHyper,
+      Mux(vSegmentException,
+        false.B,
+        lsq.io.exceptionAddr.isHyper
+      )
+    )
+  )
```

Comment:
This should be written as `!atomicsException && (misalignBufExceptionOverwrite || ...)`

Reply:
Already resolved